### PR TITLE
Further improvements to header files and shared lib version

### PR DESCRIPTION
**Pull request by besser82**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -150,8 +156,9 @@ subdir('examples')
...
 	libraries: lib_wlr,
-	version: '0.0.1',
-	filebase: 'wlroots',
-	name: 'wlroots',
+	subdirs: '@0@-@1@'.format(meson.project_name(), meson.project_version()),
+	version: meson.project_version(),
+	filebase: meson.project_name(),
+	name: meson.project_name(),
 	description: 'Wayland compositor library',
 )
```
Comment (SirCmpwn) [inline meson.build]: -1 to this commit. I'd rather not try to support several wlroots versions installed side-by-side.

**Comment by SirCmpwn** [comment]
Looks good aside from that issue.

File: meson.build (hunk 1 above)
Comment (besser82) [reply meson.build]: Well, it's less about supporting parallel installs of different versions.  The sole purpose of that part is just to have the installed include-tree in an isolated directory, so it won't (possibly) clash with the include-tree in the source-dir.

Comment (SirCmpwn) [reply meson.build]: Still not convinced. I don't like it.

Comment (besser82) [reply meson.build]: So how should that be changed?  Just keeping `meson.project_name()` for the subdir?

Comment (SirCmpwn) [reply meson.build]: Aye.

Comment (besser82) [reply meson.build]: Fixed.

**Comment by SirCmpwn** [comment]
LGTM, thanks! Will merge on build success.

**Comment by besser82** [comment]
Yay!  =)

**Comment by besser82** [comment]
@SirCmpwn Ready to merge, now  =)

**Comment by SirCmpwn** [comment]
Thanks!

**Comment by besser82** [comment]
You're welcome!

**Pull request merged**